src/lm/tools/copilotRemoteAgentTool.ts

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+
+import * as vscode from 'vscode';
+import { CopilotRemoteAgentManager } from '../../github/copilotRemoteAgent';
+
+export interface CopilotRemoteAgentToolParameters {
+	// The LLM is inconsistent in providing repo information.
+	// For now, we only support the active repository in the current workspace.
+	// repo?: {
+	// 	owner?: string;
+	// 	name?: string;
+	// };
+	title: string;
+	body?: string;
+}
+
+export class CopilotRemoteAgentTool implements vscode.LanguageModelTool<CopilotRemoteAgentToolParameters> {
+	public static readonly toolId = 'github-pull-request_copilot-coding-agent';
+
+	constructor(private manager: CopilotRemoteAgentManager) { }
+
+	async prepareInvocation(options: vscode.LanguageModelToolInvocationPrepareOptions<CopilotRemoteAgentToolParameters>): Promise<vscode.PreparedToolInvocation> {
+		const { title } = options.input;
+		const targetRepo = await this.manager.repoInfo();
+		const autoPushEnabled = this.manager.autoCommitAndPushEnabled();
+		return {
+			invocationMessage: vscode.l10n.t('Launching remote coding agent...'),
+			confirmationMessages: {
+				message: targetRepo && autoPushEnabled
+					? vscode.l10n.t('The coding agent will continue work on "**{0}**" in a new branch on "**{1}/{2}**". Any uncommitted changes will be **automatically pushed to your default remote ({3})** and included.', title, targetRepo.owner, targetRepo.repo, targetRepo.remote)
+					: vscode.l10n.t('The coding agent will start working on "**{0}**"', title),
+				title: vscode.l10n.t('Start remote coding agent?'),
+			}
+		};
+	}
+
+	async invoke(
+		options: vscode.LanguageModelToolInvocationOptions<CopilotRemoteAgentToolParameters>,
+		_: vscode.CancellationToken
+	): Promise<vscode.LanguageModelToolResult | undefined> {
+		const title = options.input.title;
+		const body = options.input.body || '';
+		const targetRepo = await this.manager.repoInfo();
+		if (!targetRepo) {
+			return new vscode.LanguageModelToolResult([
+				new vscode.LanguageModelTextPart(vscode.l10n.t('No repository information found. Please open a workspace with a Git repository.'))
+			]);
+		}
+		const result = await this.manager.invokeRemoteAgent(title, body);
+		if (result.state === 'error') {
+			throw new Error(result.error);
+		}
+		return new vscode.LanguageModelToolResult([
+			new vscode.LanguageModelTextPart(result.llmDetails || result.link)
```

Comment:
Same as in `CopilotRemoteAgentManager`, it would be cool to have this link open the PR webview.

Reply:
Not possible yet to embed deeplink here, writing up that feature request...

<img width="465" alt="Image" src="https://github.com/user-attachments/assets/eedaca9d-ec09-4642-92c8-18ee3e89521a" />